MP-5705: marketplace wishlist (#289)
spryker/spryker-docs: docs/marketplace/dev/feature-walkthroughs/202108.0/marketplace-wishlist-feature-walkthrough.md

With the *Marketplace Wishlist* feature, the customers can track and save merchant offers and products they wish to purchase through the wishlist. The customers can create multiple wishlists and customize their names.

## Modules diagram

![Entity diagram](https://confluence-connect.gliffy.net/embed/image/948f1aa1-e020-406e-b6b4-6e5e6cd2a5a5.png?utm_medium=live&utm_source=confluence)

**Modules**

**MerchantProductOfferWishlist** - extends `WishlistItem` DB table with new column.

**MerchantProductWishlist** - extends `WishlistItem` DB table with new column.

**WishlistPageExtension** - contains plugin interfaces for extending functionality of `WishlistPage` module.

> **MerchantWidget** - created new `SoldByMerchantWidget` to show owners of merchant products or product offers.

Reviewer: Is it in the feature at all?

**WishlistWidget** - created new molecules for PDP and deprecated old one.

> **ProductDetailPage** - replace molecules by new from `WishlistWidget`.

Reviewer: not in the feature

**MerchantProductOfferWidget** - created new plugins and molecules to extend forms and request data with `merchant_reference` and `product_offer_reference` on PDP.

**MerchantProductWidget** - created new plugins and molecules to extend forms and request data with `merchant_reference` on PDP.

**Wishlist** - included plugins for extending functionality of the Wishlist with supporting merchant products and product offers.

**WishlistPage** - adjusted functionality to show all information about merchants and product offers of products.

> **MerchantSwitcher** - created new plugins to support WishList.

Reviewer: same here

Reviewer: please only describe the relevant to the feature modules (see release app)

| INTEGRATION GUIDES | GLUE API GUIDES  |
| ------------- | -------------- |
| [Marketplace Wishlist feature integration](/docs/marketplace/dev/feature-integration-guides/{{page.version}}/marketplace-wishlist-feature-integration.html) | [Managing wishlists](/docs/marketplace/dev/glue-api-guides/{{page.version}}/wishlists/managing-wishlists.html)
| [Glue API: Marketplace Product Offer + Wishlist feature integration](/docs/marketplace/dev/feature-integration-guides/{{page.version}}/glue/marketplace-product-offer-wishlist-feature-integration.html) | [Managing wishlist items](/docs/marketplace/dev/glue-api-guides/{{page.version}}/wishlists/managing-wishlist-items.html) |

> ## Marketplace Wishlist - Modules diagram

Reviewer: ```suggestion
## Module dependency graph
```

![Entity diagram](https://confluence-connect.gliffy.net/embed/image/e7a2ef43-7eb8-435a-870b-d8012fe8bd07.png?utm_medium=live&utm_source=confluence)

**MerchantProductWishlist** - Provides ability to work with merchant product in wishlist.

**MerchantProductOfferWishlist** - Provides ability to work with product offers in wishlist.

**Wishlist** - Provides infrastructure and functionality to handle multiple wishlists for a customer account.

> **WishlistPage** - Provides wishlist management functionality.

Reviewer: ```suggestion
**WishlistPage** - Provides enduser wishlist management functionality.
```

**WishlistWidget** - Provides wishlist rendering functionality integration to other modules.

## Marketplace Wishlist - Domain model

> The following schema illustrates relations in the Marketplace Wishlist entity:

Reviewer: ```suggestion
The following diagram illustrates domain model of Marketplace Wishlist feature:
```
